src/kudu/consensus/leader_election.h

```diff
@@ -97,37 +120,131 @@ class VoteCounter {
 
 class FlexibleVoteCounter : public VoteCounter {
  public:
-  FlexibleVoteCounter(const LastKnownLeader& last_known_leader, RaftConfigPB config);
+  FlexibleVoteCounter(
+      const int64_t election_term, const LastKnownLeader& last_known_leader,
+      RaftConfigPB config);
 
   // Synchronization is done by the LeaderElection class. Therefore, VoteCounter
   // class doesn't need to take care of thread safety of its book-keeping
   // variables.
   Status RegisterVote(
-      const std::string& voter_uuid, ElectionVote vote,
+      const std::string& voter_uuid, VoteInfo vote,
       bool* is_duplicate) override;
   bool IsDecided() const override;
   Status GetDecision(ElectionVote* decision) const override;
  private:
+  // Mapping from UUID to an iterator in its voter history.
+  typedef std::map<std::string, std::vector<PreviousVotePB>::const_iterator>
+  VoteHistoryIteratorMap;
+
+  // Mapping from region to set of voter UUIDs that have responded in that
+  // region.
+  typedef std::map<std::string, std::set<std::string> > RegionToVoterSet;
+
+  // UUID and term pair for which a vote was given.
+  typedef std::pair<std::string, int64_t> UUIDTermPair;
```

Comment:
this typedef can be serviced by something like PreviousVotePB, which has the same 2 fields.

Reply:
Or vice-versa